pkg/healthcheck/healthcheck.go

```diff
@@ -2829,3 +2826,15 @@ func containsProxy(pod corev1.Pod) bool {
 	}
 	return false
 }
+
+// shouldSkip returns true if a Pod should be skipped when validating status
+// as it would not have a proxy container running.
+func shouldSkip(status string) bool {
```

Comment:
It seemed better to keep a list of statuses we want to skip rather than growing the conditional.

To deal with a list of statuses—and not introduce a loop label—I moved things it into a separate function.